armory/metrics/task.py

```diff
@@ -818,6 +828,107 @@ def object_detection_mAP(y_list, y_pred_list, iou_threshold=0.5, class_list=None
     return np.fromiter(ap_per_class.values(), dtype=float).mean()
 
 
+def armory_to_tide_ground_truth(y_dict, image_id, data_ground_truth):
+    for y in [dict(zip(y_dict, t)) for t in zip(*y_dict.values())]:
+        x1, y1, x2, y2 = y["boxes"]
+        width = abs(x1 - x2)
+        height = abs(y1 - y2)
+        x_min = min(x1, x2)
+        y_min = min(y1, y2)
+
+        y_tidecv = {
+            "image_id": image_id,
+            "class_id": y["labels"],
+            "box": [x_min, y_min, width, height],
+        }
```

Comment:
I think it might be easier to test / understand if the guts of this translation was its own function, and then the looping and adding to the ground truth or detection can happen in the main function.

Reply:
resolved with [8755a06](https://github.com/twosixlabs/armory/pull/1762/commits/8755a0630229a50d246025b27234fbcda057ece7)